app/models/mixins/authentication_mixin.rb

```diff
@@ -226,7 +255,21 @@ def authentication_check_types(*args)
     # Let the individual classes determine what authentication(s) need to be checked
     types = authentications_to_validate if self.respond_to?(:authentications_to_validate) && types.nil?
     types = [nil] if types.blank?
-    types.to_miq_a.each { |t| authentication_check(t, options) }
+    types.to_miq_a.each do |t|
+      success = authentication_check(t, options).first
+      retry_scheduled_authentication_check(t, options) unless success
+    end
+  end
+
+  def retry_scheduled_authentication_check(t, options)
+    return unless options[:attempt]
+    auth = authentication_best_fit(t)
```

Comment:
authentication_check doesn't return you the failed/successful authentication row, just the result (good/bad) and the error message.  We have to look up the authentication of the failed attempt so we can check if it's a retryable status.